src/elm/Msg.elm

```diff
@@ -0,0 +1,123 @@
+module Msg exposing (Interval(..), Msg(..), RefreshData)
+
+import Alerts exposing (Alert(..))
+import Api.Pagination as Pagination
+import Browser exposing (Document, UrlRequest)
+import Browser.Dom as Dom
+import Browser.Events exposing (Visibility(..))
+import Errors exposing (Error)
+import Http
+import Http.Detailed
+import Pages.Build.Model
+import Pages.Pipeline.Model
+import Pages.Secrets.Model
+import Routes
+import Time exposing (Posix, Zone)
+import Toasty as Alerting
+import Vela
+    exposing
+        ( Build
+        , BuildNumber
+        , Builds
+        , CurrentUser
+        , Engine
+        , Event
+        , Field
+        , FocusFragment
+        , Hooks
+        , Log
+        , Org
+        , Repo
+        , Repositories
+        , Repository
+        , Secret
+        , Secrets
+        , Session
+        , SourceRepositories
+        , Step
+        , StepNumber
+        , Steps
+        , Theme
+        , Type
+        , User
+        )
+
+
+type Interval
+    = OneSecond
+    | OneSecondHidden
+    | FiveSecond RefreshData
+    | FiveSecondHidden RefreshData
+
+
+type alias RefreshData =
+    { org : Org
+    , repo : Repo
+    , build_number : Maybe BuildNumber
+    , steps : Maybe Steps
+    }
+
+
+type Msg
```

Comment:
by moving Msg into its own Module, we no longer have to pass generic Msgs down 15 function calls or bloat partial models with Msg aliases.
Now, we can import Msg(..) and use ToggleFavorite on ANY repo page